### PR TITLE
Add BOARDCTL_SWITCH_BOOT to support switch boot system

### PR DESCRIPTION
**Pull request by liuguo09**
## Summary

board/ctrl: add BOARDCTL_SWITCH_BOOT switch boot system support

BOARDIOC_SWITCH_BOOT is required to communicate the boot partition from
userspace (OTA subsystem) to board. It can be used to change the system
boot behavior. So it's useful for A/B boot or even single boot case.

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Comment by patacongo** [comment]
Seems rather application-specific.  Have you considered the generic BOARDIOC_IOCTL instead?Sent from my Galaxy

**Comment by xiaoxiang781216** [comment]
> Seems rather application-specific.  Have you considered the generic BOARDIOC_IOCTL instead?Sent from my Galaxy

Yes, but it isn't too specific. If the board support A/B boot, something like BOARDIOC_SWITCH_BOOT is required to commuicate the boot partition from userspace(OTA subsystem) to board, so it's a common requirement for A/B boot:

1. Download the new image from the sever
2. Write the new image to the alternative partition
3. Call BOARDCTL_SWITCH_BOOT to the alternative system

This is even useful in the single boot partition case, if we want to implement the reliable update:

1. Download the new image from the sever
2. Call BOARDCTL_SWITCH_BOOT and reboot to a small system
3. Write the new image to the main partition in the small system
4. Call BOARDCTL_SWITCH_BOOT again to the main system

You can see that it's key point to allow updater switch the boot partition.

File: boards/Kconfig [hunk 1]
```diff
@@ -3389,6 +3389,13 @@ config BOARDCTL_UNIQUEKEY_SIZE
 		Provides the size of the memory buffer that must be provided by the
 		caller of board_uniquekey() in which to receive the board unique KEY.
 
+config BOARDCTL_SWITCH_BOOT
+	bool "Switch Boot A/B System"
+	default n
+	---help---
+		Enables support for the BOARDIOC_SWITCH_BOOT boardctl() command.
+		Switch to the updated or the specified Boot A/B system.
+
```
Comment (davids5) [inline boards/Kconfig:3398]: This is to vague. Please add more information:

This API can be used to change the system boot behavior.  For instance, once a firmware updated has completed successfully this boardctl can be used modify FLASH bank selection.

Comment (davids5) [reply boards/Kconfig:3398]: @xiaoxiang781216 The salient points from [comment](https://github.com/apache/incubator-nuttx/pull/4191#issuecomment-883941665) should be added to either the Kconfig.

Comment (liuguo09) [reply boards/Kconfig:3401]: @davids5 @xiaoxiang781216 I have updated the patch accordingly. Pls help review, thanks.

**Review by davids5 — APPROVED** [review]
This work for me.

**Comment by xiaoxiang781216** [comment]
@patacongo do you think this PR general enough to merge now?

**Pull request merged**